modules/programs/vdirsyncer.nix

```diff
@@ -0,0 +1,404 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.vdirsyncer;
+  
+  ## Type definitions
+  collection = types.either types.str (types.listOf types.str);
+  storageType = types.enum [
+    "caldav" "carddav" "http"
+    "filesystem" "singlefile"
+  ];
+
+  storage = types.submodule {
+    options = {
```

Comment:
what about localStorage vs remoteStorage ? can we really create pairs between 2 caldav for instance ?

Reply:
the storage might need an extraConfig option. For instance with the type "goole_calendar", I have to set 
```
# token_file = "~/.local/vdirsyncer"
client_id = "dsds"
client_secret = "~/.config/vdirsyncer/client_secret"

```

Reply:
> what about localStorage vs remoteStorage ? can we really create pairs between 2 caldav for instance ?

To be honest, I just assumed that this was possible, but I’m going to test it.

> the storage might need an extraConfig option. For instance with the type "goole_calendar", I have to set

Currently, you can’t create a storage of type "google_calendar" anyway because the allowed types are hardcoded and I haven’t implemented the google and ete stuff yet. It’s on the agenda :)